rust/feature-flags/src/api/request_handler.rs

```diff
@@ -435,34 +442,47 @@ fn decode_form_data(
     body: Bytes,
     compression: Option<Compression>,
 ) -> Result<FlagRequest, FlagError> {
-    #[derive(Deserialize)]
-    struct FormData {
-        data: String,
-    }
-
-    let form_data_str = String::from_utf8(body.to_vec()).map_err(|e| {
-        FlagError::RequestDecodingError(format!("Invalid UTF-8 in form data: {}", e))
-    })?;
-    let form: FormData = serde_urlencoded::from_str(&form_data_str).map_err(|e| {
-        FlagError::RequestDecodingError(format!("Failed to parse form data: {}", e))
-    })?;
-
-    let data_str = urlencoding::decode(&form.data)
-        .map_err(|e| FlagError::RequestDecodingError(format!("URL decode error: {}", e)))?
-        .into_owned();
-
     match compression {
         Some(Compression::Gzip) => Err(FlagError::RequestDecodingError(
             "Gzip compression not supported for form-urlencoded data".to_string(),
         )),
         Some(Compression::Unsupported) => Err(FlagError::RequestDecodingError(
             "Unsupported compression type".to_string(),
         )),
-        _ => {
-            let decoded_bytes = general_purpose::STANDARD.decode(data_str).map_err(|e| {
+        Some(Compression::Base64) | None => {
+            // Parse form data into a utf-8 string
+            let form_data_str = String::from_utf8(body.to_vec()).map_err(|e| {
+                FlagError::RequestDecodingError(format!("Invalid UTF-8 in form data: {}", e))
+            })?;
+
+            let form: FormData = serde_urlencoded::from_str(&form_data_str).map_err(|e| {
+                FlagError::RequestDecodingError(format!("Failed to parse form data: {}", e))
+            })?;
+
+            // URL decode first (matching Python's unquote)
+            let decoded_form = urlencoding::decode(&form.data)
+                .map_err(|e| FlagError::RequestDecodingError(format!("URL decode error: {}", e)))?
+                .into_owned();
+
+            // Remove spaces only (not all whitespace)
+            let no_spaces = decoded_form.replace(' ', "");
```

Comment:
I have to admin, I'm a little confused by this too. I thought the issue was whitespace in the decoded content, not in the base64 encoded string. Spaces would be invalid in a base64 encoded string. It wouldn't be invalid in the decoded content.

Reply:
added context on this in the latest commit